src/xmipp/libraries/reconstruction/metadata_xml.cpp

```diff
@@ -89,27 +89,28 @@ class ProgTest: public XmippProgram
             std::cout << fnIn << std::endl;
             sortedMD.sort(MD,MDL_MICROGRAPH);
 
-            sortedMD.getValue(MDL_MICROGRAPH, name, MD.firstObject());
+            sortedMD.getValue(MDL_MICROGRAPH, name, MD.firstRowId());
             nodirName=name.removeDirectories();
             nodirName=nodirName.removeAllExtensions();
             fhOut<<"<micrograph id=\""<<nodirName<<"\">"<<std::endl;
 
-            sortedMD.getValue(MDL_XCOOR, x, sortedMD.firstObject());
-            sortedMD.getValue(MDL_YCOOR, y, sortedMD.firstObject());
+            sortedMD.getValue(MDL_XCOOR, x, sortedMD.firstRowId());
+            sortedMD.getValue(MDL_YCOOR, y, sortedMD.firstRowId());
             fhOut<<"<coordinate x=\""<<x<<"\" y=\""<<y<<"\"/>"<<std::endl;
 
-            FOR_ALL_OBJECTS_IN_METADATA(sortedMD)
+            for (auto idIt = sortedMD.ids().begin(); idIt != sortedMD.ids().end(); ++idIt)
```

Comment:
`.begin()+1`?

Reply:
Well, this doesn't work, does it? `+1` is not defined on iterator.